apps/prairielearn/elements/pl-excalidraw/pl-excalidraw.js

```diff
@@ -0,0 +1,126 @@
+/* eslint-env browser,jquery */
+
+// This module is redefined in the import map with the same name
+import { ExcalidrawLib, React, ReactDOM } from '@prairielearn/excalidraw-builds';
+
+const elt = React.createElement;
+
+const Footer = ({ unsaved, readOnly }) => {
+  const desc = unsaved ? 'Unsaved' : 'Saved';
+  return elt(
+    ExcalidrawLib.Footer,
+    null,
+    readOnly
+      ? null
+      : elt(
+          ExcalidrawLib.Button,
+          {
+            'aria-label': desc,
+            type: 'button',
+            disabled: true,
+            title: desc,
+            size: 'medium',
+            className: unsaved ? 'save-button save-button-unsaved' : 'save-button',
+          },
+          desc,
+        ),
+  );
+};
+
+const DrawWidget = ({ sketchName, metadata, setHiddenInput }) => {
+  const [readOnly, setReadOnly] = React.useState(false);
+  const [unsaved, setUnsaved] = React.useState(false);
+  const [lib, setLib] = React.useState(null);
+  const [sceneVer, setSceneVer] = React.useState(0);
+
+  /* First time setup */
+  React.useEffect(() => {
+    setReadOnly(metadata.read_only);
+  }, [metadata]);
+
+  /* Autosave */
+  React.useEffect(() => {
+    // No need to autosave a read-only element
+    if (!lib || metadata.read_only) return;
+    const autoSave = setTimeout(() => {
+      setHiddenInput(
+        ExcalidrawLib.serializeAsJSON(
+          lib.getSceneElements(),
+          lib.getAppState(),
+          lib.getFiles(),
+          'local',
+        ),
```

Comment:
It's unfortunate that this hardcodes serialization with multiple lines and indentation. That whitespace will almost certainly chew through our maximum body size of 5MB. It'd be really nice if Excalidraw provided a plain `serialize` function that just returned a JS object that we could then serialize on our own.

Not requesting any changes here, unless you want to upstream such a function.